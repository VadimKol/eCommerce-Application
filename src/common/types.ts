import type { BaseAddress, CustomFieldsDraft, StoreResourceIdentifier } from '@commercetools/platform-sdk';

export interface AuthContextInterface {
  isAuthenticated: boolean;
  handleLogin: () => void;
  handleLogout: () => void;
}

export type CategoriesContextType = {
  categories: CategoriesData | null;
  loading: boolean;
  error: Error | null;
};

export interface Credentials {
  clientId: string;
  clientSecret: string;
}

export interface User {
  username: string;
  password: string;
}

export interface GeekShopCustomerDraft {
  /**
   *	Email address of the Customer that is [unique](/../api/customers-overview#customer-uniqueness) for an entire Project or Store the Customer is assigned to.
   *	It is the mandatory unique identifier of a Customer.
   *
   *
   */
  readonly email: string;
  /**
   *	Password of the Customer.
   *
   *
   */
  readonly password: string;
  /**
   *	Given name (first name) of the Customer.
   *
   *
   */
  readonly firstName?: string;
  /**
   *	Family name (last name) of the Customer.
   *
   *
   */
  readonly lastName?: string;
  /**
   *	Middle name of the Customer.
   *
   *
   */
  readonly middleName?: string;
  /**
   *	Title of the Customer, for example, 'Dr.'.
   *
   *
   */
  readonly title?: string;
  /**
   *	Salutation of the Customer, for example, 'Mr.' or 'Mrs.'.
   *
   *
   */
  readonly salutation?: string;
  /**
   *	Date of birth of the Customer.
   *
   *
   */
  readonly dateOfBirth?: string;
  /**
   *	Company name of the Customer.
   *
   *
   */
  readonly companyName?: string;
  /**
   *	Individual VAT ID of the Customer.
   *
   *
   */
  readonly vatId?: string;
  /**
   *	Addresses of the Customer.
   *
   *
   */
  readonly addresses?: BaseAddress[];
  /**
   *	Indexes of entries in `addresses` to set as shipping addresses.
   *	The `shippingAddressIds` of the [Customer](ctp:api:type:Customer) will be replaced by these addresses.
   *
   *
   */
  readonly shippingAddresses?: number[];
  /**
   *	Index of the address in the `addresses` array to use as the default shipping address.
   *	The `defaultShippingAddressId` of the Customer will be set to the `id` of that address.
   *
   *
   */
  readonly defaultShippingAddress?: number;
  /**
   *	Indexes of entries in `addresses` to set as billing addresses.
   *	The `billingAddressIds` of the [Customer](ctp:api:type:Customer) will be replaced by these addresses.
   *
   *
   */
  readonly billingAddresses?: number[];
  /**
   *	Index of the address in the `addresses` array to use as the default billing address.
   *	The `defaultBillingAddressId` of the Customer will be set to the `id` of that address.
   *
   *
   */
  readonly defaultBillingAddress?: number;
  /**
   *	Custom Fields for the Customer.
   *
   *
   */
  readonly custom?: CustomFieldsDraft;
  /**
   *	Preferred language of the Customer. Must be one of the languages supported by the [Project](ctp:api:type:Project).
   *
   *
   */
  readonly locale?: string;
  /**
   *	Sets the [Stores](ctp:api:type:Store) for the Customer.
   *
   *
   */
  readonly stores?: StoreResourceIdentifier[];
}

export type ProductDetails = {
  name?: string;
  description?: string;
  price?: string | null;
  discountedPrice?: string | null;
  currency?: string | null;
  images?: string[];
  attributes?: { name: string; value: string | number | string[] }[];
  availability?: {
    isOnStock?: boolean | null;
    availableQuantity?: number | null;
  };
};

export type Product = {
  id: string;
  name: string;
  description: string;
  slug: string;
  key: string;
  sku: string;
  quantity: number;
  price: string;
  images: string[];
  categoryId: string;
  slugCategory: string;
  keyCategory: string;
  subcategoryId: string;
  slugSubCategory: string;
  keySubCategory: string;
};

export type Subcategory = {
  id: string;
  name: string;
  slug: string;
  key: string;
  products: Product[];
};

export type Category = {
  id: string;
  name: string;
  slug: string;
  key: string;
  subcategories: Subcategory[];
};

export type CategoriesData = Category[];

export type UrlParams = {
  categoryName?: string;
  subcategoryName?: string;
  productName?: string;
};

export type LoaderData = {
  category?: Category;
  subcategory?: Subcategory;
  product?: Product;
};
