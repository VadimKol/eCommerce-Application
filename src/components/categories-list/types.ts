import type { Category } from '@/common/types';

export interface CategoriesListProps {
  categories: Category[];
}
