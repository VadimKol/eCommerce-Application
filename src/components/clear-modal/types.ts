import { type LineItem } from '@commercetools/platform-sdk';

export interface ClearModalProps {
  isModal: boolean;
  setIsModal: (isModal: boolean) => void;
  cartItems: LineItem[];
  clearFromCart: (lineItems: string[]) => Promise<void>;
}
