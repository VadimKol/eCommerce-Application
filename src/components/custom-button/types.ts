export interface ButtonProps {
  children: React.ReactNode;
  type?: 'submit' | 'reset' | 'button';
  variant?: 'primary' | 'secondary' | 'tertiary' | 'cancel';
  onClick?: VoidFunction;
  className?: string;
  isDisabled?: boolean;
  id?: string;
}
