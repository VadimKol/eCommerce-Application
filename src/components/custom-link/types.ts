export interface LinkProps {
  children: React.ReactNode;
  to: string;
  variant?: 'primary' | 'secondary' | 'tertiary' | 'cancel';
  className?: string;
  isDisabled?: boolean;
  id?: string;
}
