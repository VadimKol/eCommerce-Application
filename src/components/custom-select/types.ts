export interface SelectProps {
  selectItems: string[];
  selectState: string;
  setSelectState: (selectState: string) => void;
  className?: string;
  setLoadingProducts: (loadingProducts: boolean) => void;
}
