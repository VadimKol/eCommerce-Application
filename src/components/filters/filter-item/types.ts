import { type CatalogAction } from '@/common/types';

export interface FilterItemProps {
  priceSlider: boolean;
  checkboxesList: boolean[];
  stateField: string;
  dispatch: React.Dispatch<CatalogAction>;
  priceFilter?: [number, number];
}
