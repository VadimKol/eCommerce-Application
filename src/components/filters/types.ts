export interface FilterProps {
  priceFilter: [number, number];
  setPriceFilter: (priceFilter: [number, number]) => void;
  franchises: boolean[];
  setFranchises: (franchises: boolean[]) => void;
  setPage: (page: number) => void;
}
