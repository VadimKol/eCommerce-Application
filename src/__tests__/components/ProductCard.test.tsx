import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { type Product } from '@/common/types';
import { ProductCard } from '@/components/product-card/ProductCard';

it('renders correctly', () => {
  const product: Product = {
    id: '',
    name: '',
    description: '',
    slug: '',
    key: '',
    sku: '',
    quantity: 0,
    price: '00.00',
    discount: '00.00',
    images: [''],
    categories: [],
    keyCategory: '',
    keySubCategory: '',
  };

  const { container } = render(
    <BrowserRouter>
      <ProductCard product={product} categoryName="Cosplay" subcategoryName="Mask" />
    </BrowserRouter>,
  );

  expect(container).toMatchSnapshot();
});
