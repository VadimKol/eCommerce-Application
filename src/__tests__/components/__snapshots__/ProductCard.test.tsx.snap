// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="image_container"
    >
      <div
        class="loader"
      />
      <img
        alt="Product"
        class="product_img"
        src=""
        style="display: none;"
      />
    </div>
    <div
      class="text_container"
    >
      <a
        class="card_link"
        href="/catalog/Cosplay/Mask/"
      >
        <h2
          class="card_title"
        />
      </a>
      <p
        class="card_description"
      />
      <p
        class="price_block"
      >
        <span>
          $
          00.00
        </span>
        <span
          class="product_discount"
        >
          $
          00.00
        </span>
      </p>
      <button
        class="button button--tertiary btn_cart"
        disabled=""
        id=""
        type="button"
      >
        Loading...
      </button>
    </div>
  </div>
</div>
`;
